### PR TITLE
Merge all the Custom CSS changes done in the 4.4.2 release into master.

### PR DESCRIPTION
**Pull request by georgestephanis**

### CODE REVIEW

**Comment by dereksmart** [comment]
Looks fine, tests as well as branch-4.4 does

**Comment by dereksmart** [comment]
I'm going to merge this now before the next push causes Travis to fail, and this is blocking me from building a new master-stable.  @georgestephanis can you please look at that TODO once it's merged?

File: modules/custom-css.php [hunk 1]
```diff
@@ -13,7 +13,35 @@
...
+	// @see https://core.trac.wordpress.org/changeset/38829
+	if ( function_exists( 'wp_get_custom_css' ) ) {
+		if ( ! function_exists( 'wp_update_custom_css_post' ) ) {
+			wp_die( 'Please run a SVN up to get the latest version of trunk, or update to at least 4.7 RC1' );
+		}
+		if ( ! Jetpack_Options::get_option( 'custom_css_4.7_migration' ) ) {
+			include_once dirname( __FILE__ ) . '/custom-css/migrate-to-core.php';
+		}
+
+		// TODO: DELETE THIS
```
Comment (georgestephanis) [inline modules/custom-css.php]: Yeah, not a big deal, and could be helpful if anyone has to revert or re-import.

**Comment by iandunn** [comment]
Unfortunately, 4.4.2 broke the some functionality of the [WordCamp Site Cloner](https://meta.trac.wordpress.org/browser/sites/trunk/wordcamp.org/public_html/wp-content/plugins/wordcamp-site-cloner?rev=4501) and [WordCamp Remote CSS](https://meta.trac.wordpress.org/browser/sites/trunk/wordcamp.org/public_html/wp-content/plugins/wordcamp-remote-css?rev=4501) plugins, because they rely on the Custom CSS module.

I'm having trouble figuring out if the break with back-compat was intentional or not, though. [r1544849-plugins](https://plugins.trac.wordpress.org/changeset/1544849/jetpack/trunk/modules/custom-css) removed the call to `Jetpack_Custom_CSS::init`. It also added a bunch of files, even though they're not being used yet. It seems like the `init` call was supposed to remain, since [it exists in George's PR](https://github.com/Automattic/jetpack/compare/7cbf84f1087d3bd73f701e4ccdf591d403a1e727...1e290ec646f2cf9e8484e93919826892b6382747#diff-ad3b20e74d210c1dd46dc0af3f1f9db8R44).

I may just be reading this all wrong, though; it's been a long week.

@zinigor, @georgestephanis, would you mind helping me understand? If the break was intentional, I'm happy to refactor my plugins to work with the new version.

**Comment by iandunn** [comment]
Doh, I missed that there's a new `modules/custom-css.php` outside the `custom-css` folder, and the `init` call was moved there. That makes it seem like the back-compat break was intention, so I'll move forward with updating my plugins.

**Pull request merged**